tests/test.c

```diff
@@ -4507,17 +4507,18 @@ void test_kafka_topics(const char *fmt, ...) {
         int r;
         va_list ap;
         test_timing_t t_cmd;
-        const char *kpath, *zk;
```

Comment:
zk is still required for older Kafka versions, which we continue to test with, so this needs some more logic.

Reply:
I've tested it locally with Kafka v2.7.0 and it runs fine (tried even v2.5.0 and still works, see bellow[1]), I think the problem is with the static size buffer for [cmd](https://github.com/edenhill/librdkafka/blob/master/tests/test.c#L4506).

Here https://app.travis-ci.com/github/edenhill/librdkafka/jobs/557875940#L8186 it looks like the command is not complete, there is a missing setting for `--config delete.retention.ms`, the `test_kafka_topics()` should make boundary check for cmd since the `kpath` and input `fmt` are of variable length

>zk is still required for older Kafka versions

what is the oldest version you want to support if there would be a still problem with backwards capability, we can create `cmd` dynamically based on the `TEST_KAFKA_VERSION` env variable?

[1]:
```bash
bash-3.2$ TESTS=0077 make
Running tests in parallel
./run-test.sh
############## ./test-runner ################
### Running test ./test-runner in bare mode ###
[<MAIN>                      /  0.000s] Setting test timeout to 10s * 1.0
[<MAIN>                      /  0.000s] Git version: a98d26e5@master
[<MAIN>                      /  0.000s] Broker version: 2.5.0 (2.5.0.0)
[<MAIN>                      /  0.000s] Tests to run : 0077
[<MAIN>                      /  0.000s] Test mode    : bare
[<MAIN>                      /  0.000s] Test scenario: default
[<MAIN>                      /  0.000s] Test filter  : no filter
[<MAIN>                      /  0.000s] Test timeout multiplier: 2.7
[<MAIN>                      /  0.000s] Action on test failure: continue other tests
[<MAIN>                      /  0.000s] Current directory: /Users/lada/Workspace/librdkafka/tests
[<MAIN>                      /  0.000s] Setting test timeout to 30s * 2.7
[<MAIN>                      /  0.000s] 1 test(s) running: 0077_compaction
[0077_compaction             /  0.000s] ================= Running test 0077_compaction =================
[0077_compaction             /  0.000s] ==== Stats written to file stats_0077_compaction_4265141123287969475.json ====
[0077_compaction             /  0.000s] Using topic "rdkafkatest_rnd78f2c57b0a834c46_0077-compaction.c"
[0077_compaction             /  0.000s] Test compaction on topic rdkafkatest_rnd78f2c57b0a834c46_0077-compaction.c with no compression (40 messages)
[0077_compaction             /  0.000s] Executing: /Users/lada/Workspace/librdkafka/tests/tmp/LibrdkafkaTestCluster/KafkaBrokerApp/kafka/2.5.0/bin/kafka-topics.sh --bootstrap-server PLAINTEXT://0.0.0.0:50808,PLAINTEXT://0.0.0.0:50811,PLAINTEXT://0.0.0.0:50814 --create --topic "rdkafkatest_rnd78f2c57b0a834c46_0077-compaction.c" --partitions 1 --replication-factor 1 --config cleanup.policy=compact --config segment.ms=10000 --config segment.bytes=10000 --config min.cleanable.dirty.ratio=0.01 --config delete.retention.ms=86400 --config file.delete.delay.ms=100
WARNING: Due to limitations in metric names, topics with a period ('.') or underscore ('_') could collide. To avoid issues it is best to use either, but not both.
[<MAIN>                      /  1.005s] 1 test(s) running: 0077_compaction
Created topic rdkafkatest_rnd78f2c57b0a834c46_0077-compaction.c.
[0077_compaction             /  1.483s] exec: duration 1483.113ms
[0077_compaction             /  1.486s] Setting test timeout to 120s * 2.7
[0077_compaction             /  1.502s] Created    kafka instance 0077_compaction#producer-1
[0077_compaction             /  1.502s] Producing 20 messages (total of 20000 bytes) of compactable messages
[0077_compaction             /  1.503s] Created    kafka instance 0077_compaction#producer-2
[0077_compaction             /  1.503s] 0077_compaction#producer-2: Flushing 20 messages
[0077_compaction             /  1.593s] FLUSH: duration 89.913ms
[0077_compaction             /  1.593s] Producing 40 messages for 4 keys
[0077_compaction             /  1.639s] PRODUCE.DELIVERY.WAIT: duration 45.441ms
[0077_compaction             /  1.639s] Producing 20 messages (total of 20480 bytes) of compactable messages
[0077_compaction             /  1.639s] Created    kafka instance 0077_compaction#producer-3
```
[1]: https://github.com/edenhill/librdkafka/blob/master/tests/test.c#L4506

Reply:
We support Kafka 0.8 and later.